### PR TITLE
fix(react): withRef component displayNames

### PR DESCRIPTION
**Pull request by rabelloo**
## Purpose

Closes https://github.com/onfido/castor/issues/151

## Approach

Explicitly name functions used inside `withRef` so that the function's name is used as `displayName`.

## Testing

Go to Storybook, select their stories, click show code, confirm it shows the component name.

## Risks

There's no way to guarantee `withRef` will be used with named functions in the future.
Anonymous functions will continue to have the same issue.
There is a lint rule for `forwardRed`, but then the types get all messed up (which `withRef` fixes).

### CODE REVIEW

**Comment by josokinas** [comment]
I think the solution here is fine, however how do you feel about setting `displayName` for when component uses `withRef`, e.g.:

```js
Textarea.displayName = 'Textarea';
```

**Review by josokinas — APPROVED** [review]

**Comment by rabelloo** [comment]
> how do you feel about setting `displayName` for when component uses `withRef`

It'd require some more type juggling, plus the name isn't localised in a single line so it could make it easier to miss, but I don't mind that solution either.

I'll push another commit for comparison and you tell me what you think.

**Comment by josokinas** [comment]
> I'll push another commit for comparison and you tell me what you think.

Ah, not needed - I'm fine with this! Thanks 😏

**Comment by rabelloo** [comment]
> Ah, not needed - I'm fine with this! Thanks 😏

Too late, take a look now 🤣

File: packages/react/src/utils/withRef.ts [hunk 1]
```diff
@@ -6,6 +6,10 @@ import { FC, forwardRef } from 'react';
  * @param component Component to `forwardRef`.
  */
 /* eslint-disable @typescript-eslint/no-explicit-any */
-export const withRef = <C extends FC<any>>(component: C): C =>
+export const withRef = <C extends FC<any>>(component: C): C & Named =>
   forwardRef(component as any) as any;
 /* eslint-enable @typescript-eslint/no-explicit-any */
+
+interface Named {
+  displayName?: string;
```
Comment (josokinas) [inline packages/react/src/utils/withRef.ts:14]: Could this be not optional?

**Review by josokinas — APPROVED** [review]
Nice! I like with `displayName` more, but happy with both - you choose!

File: packages/react/src/utils/withRef.ts (hunk 1 above)
Comment (rabelloo) [reply packages/react/src/utils/withRef.ts:14]: It could, but it wouldn't be accurate, because the return from `withRef` has an undefined `displayName`.

**Comment by rabelloo** [comment]
> Nice! I like with `displayName` more, but happy with both - you choose!

I'll leave as is, with this we can use either in the future.

**Pull request merged**